src/vs/workbench/api/browser/mainThreadTerminalService.ts

```diff
@@ -89,14 +87,27 @@ export class MainThreadTerminalService implements MainThreadTerminalServiceShape
 			this._proxy.$initEnvironmentVariableCollections(serializedCollections);
 		}
 
-		this._terminalService.extHostReady(extHostContext.remoteAuthority!); // TODO@Tyriar: remove null assertion
+		remoteAgentService.getEnvironment().then(async env => {
```

Comment:
I think `getEnvironment()` can return null? https://github.com/microsoft/vscode/blob/39bd9df739ffd09ce5631757bb12daa14b77c403/src/vs/workbench/services/remote/common/abstractRemoteAgentService.ts#L54-L56

Reply:
Yep, this is unexpected for remotes but we fallback to the renderer's os (`OS`) so that should be fine